crates/sdk/src/tx.rs

```diff
@@ -3524,6 +3582,38 @@ pub async fn build_shielding_transfer<N: Namada>(
             .ok_or(Error::Other("Combined transfer overflows".to_string()))?;
     }
 
+    for args::TxTransparentTarget {
+        target: sus_fee_target,
+        token: sus_fee_token,
+        amount: sus_fee_amt,
+    } in &args.frontend_sus_fee
+    {
+        // Validate the amount given
+        let validated_fee_amount = validate_amount(
+            context,
+            sus_fee_amt.to_owned(),
+            sus_fee_token,
+            args.tx.force,
+        )
+        .await?;
+
+        // Decrease the amount to shield to account for this frontend fee, take
+        // it from the first shielding input that matches the fee token
+        for (_, token, amount) in &mut transfer_data.sources {
```

Comment:
hold up. So we are diverting some of the shielding funds to the fee? How did it get bundled in there when it is explicitly fed in as a separate amount. Does it simplify balance checks? Does that mean the input shielded amount is supposed to `intended_amount + fee`? with the understanding that the fee gets subtracted off?